spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -204,6 +220,36 @@ def __init__(self, path, electrical_series_name='ElectricalSeries'):
                     'end_frame': self.time_to_frame(row['stop_time'])}
                     for _, row in df_epochs.iterrows()}
 
+            # Metadata dictionary - useful for constructing a nwb file
+            self.nwb_metadata = dict()
+            self.nwb_metadata['NWBFile'] = {
+                        'session_description': nwbfile.session_description,
+                        'identifier': nwbfile.identifier,
+                        'session_start_time': nwbfile.session_start_time,
+                        'institution': nwbfile.institution,
+                        'lab': nwbfile.lab
+            }
+            self.nwb_metadata['Ecephys'] = dict()
+            # Update metadata with Device info
+            self.nwb_metadata['Ecephys']['Device'] = []
+            for dev in nwbfile.devices:
+                self.nwb_metadata['Ecephys']['Device'].append({'name': dev})
+            # Update metadata with ElectrodeGroup info
+            self.nwb_metadata['Ecephys']['ElectrodeGroup'] = []
+            for k, v in nwbfile.electrode_groups.items():
+                self.nwb_metadata['Ecephys']['ElectrodeGroup'].append({
+                    'name': v.name,
+                    'description': v.description,
+                    'location': v.location,
+                    'device': v.device.name
+                })
+            # Update metadata with ElectricalSeries info
+            self.nwb_metadata['Ecephys']['ElectricalSeries'] = []
+            self.nwb_metadata['Ecephys']['ElectricalSeries'].append({
+                'name': es.name,
+                'description': es.description
+            })
+
```

Comment:
I would make this section its own static function

Reply:
done on 1242f2c